perun-core/src/main/java/cz/metacentrum/perun/core/impl/UsersManagerImpl.java

```diff
@@ -846,31 +849,40 @@ public List<User> getUsersByAttribute(PerunSession sess, Attribute attribute, bo
 
 	@Override
 	public List<User> getUsersByAttributeValue(PerunSession sess, AttributeDefinition attributeDefinition, String attributeValue) {
-		String value = "";
+		MapSqlParameterSource namedParams = new MapSqlParameterSource();
+		namedParams.addValue("attr_id", attributeDefinition.getId());
 		String operator = "=";
-		if (attributeDefinition.getType().equals(String.class.getName())) {
-			value = attributeValue.trim();
-			operator = "=";
-		} else if (attributeDefinition.getType().equals(Integer.class.getName())) {
-			value = attributeValue.trim();
-			operator = "=";
-		}  else if (attributeDefinition.getType().equals(Boolean.class.getName())) {
-			value = attributeValue.trim();
-			operator = "=";
+		String value_part = ":value";
+
+		String attributeType = attributeDefinition.getType();
+		if (attributeType.equals(String.class.getName())) {
+			namedParams.addValue("value", attributeValue.trim());
+		} else if (attributeType.equals(Integer.class.getName())) {
+			namedParams.addValue("value", attributeValue.trim());
+		} else if (attributeType.equals(Boolean.class.getName())) {
+			namedParams.addValue("value", attributeValue.trim());
 		} else if (attributeDefinition.getType().equals(ArrayList.class.getName())) {
-			value = "%" + attributeValue.trim() + "%";
+			// escape backslashes from delimiters escaping, use delimiters to match value
 			operator = "like";
+			String escapedValue = AttributesManagerBlImpl.escapeListAttributeValue(attributeValue.trim()).replace("\\", "\\\\");
+			value_part = "any(array[(:regex1),(:regex2)])";
```

Comment:
Just a nitpick: could you please rename the regex parameters or add a comment explaining what the regexes should represent (first element match / any other element)? It helped me understand the code quickly when I saw it for the first time.